theano/sandbox/cuda/tests/test_dnn.py

```diff
@@ -230,6 +230,25 @@ def pool_2d_i2n(input, ds=(2, 2), strides=None, pad=(0, 0),
                                           output_width, output_height))
     return pooled_output
 
+def test_pooling2():
+    if not cuda.dnn.dnn_available():
+        raise SkipTest(cuda.dnn.dnn_available.msg)
+
+    x = T.ftensor4()
+    ws = T.ivector()
+    stride = T.ivector()
+    pad = T.ivector()
+    
+    pooled_x = cuda.dnn.dnn_pool(x, ws, stride=stride, pad=pad)
+    
+    f = theano.function([x, ws, stride, pad], pooled_x)
+    x_ = numpy.arange(500).reshape(1, 5, 10, 10).astype('float32')
+    ws_ = numpy.array([1, 1]).astype('int32')
+    stride_ = numpy.array([1, 1]).astype('int32')
+    pad_ = numpy.array([0, 0]).astype('int32')
+
+    print f(x_, ws_, stride_, pad_)
```

Comment:
make the test not print something and compare again the expected value. for this, you can use the function  pool_2d_i2n() defined in that file.
Call f twice with different value of ws, stride and pad, to be sure we can change them at run time.